src/components/Main/Main.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+
+import { FilmList } from "../FilmList";
+import { FilterMoviesList } from "../FilterMoviesList";
+import { genreList } from "../../films";
+import { Film, Modal } from "../../interfaces";
+import { CONSTANTS } from "../../constants";
+
+import "./style.scss";
+import { Form } from "../Form";
+import { filterByGenre, sorting } from "../../helpers";
+import { ErrorBoundary } from "../ErrorBoundary";
+
+interface MainProps {
+  modalState: Modal;
+  closeModal(): void;
+  movies: Film[];
+  openModal(type: string, filmID: string): void;
+}
+
+const Main: React.FC<MainProps> = ({
+  modalState,
+  closeModal,
+  movies,
+  openModal,
+}) => {
+  const [selectedGenre, setGenre] = useState(
+    CONSTANTS.DEFAULT_FILTERS.defaultGenre
+  );
+  const [sortBy, setSort] = useState("");
+
+  const onChangeGenre = (event: React.MouseEvent) => {
+    setSort("");
+    setGenre((prevGenre) =>
+      (event.target as HTMLInputElement).id
+        ? (event.target as HTMLInputElement).id
+        : prevGenre
+    );
+  };
+
+  const chengeSort = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setGenre(CONSTANTS.DEFAULT_FILTERS.defaultGenre);
+    setSort(event.target.value);
+  };
+
+  return (
+    <div className="main">
+      <FilterMoviesList
+        selectedGenre={selectedGenre}
+        genreList={genreList}
+        onChangeGenre={onChangeGenre}
+        chengeSort={chengeSort}
+      />
+      <ErrorBoundary>
+        <FilmList
+          movies={filterByGenre(sorting(movies, sortBy), selectedGenre)}
+          openModal={openModal}
+        />
+      </ErrorBoundary>
+      {modalState.isOpen ? (
+        <Form closeModal={closeModal} modalState={modalState} />
+      ) : (
+        <></>
+      )}
```

Comment:
why have you chosen this way, and not a portal?
if i open your modal l can't scroll your it (imagine if there is more content), on smaller screen height the part of the content on your modal is unreachable, it just can't fit, that is why i need scroll. 
Beside that, the background is scrollable, it is not a best practice.